stdlib/public/core/StaticString.swift

```diff
@@ -172,20 +172,7 @@ public struct StaticString {
       return body(UnsafeBufferPointer(
         start: utf8Start, count: utf8CodeUnitCount))
     } else {
-      var buffer: UInt64 = 0
-      var i = 0
-      let sink: (UInt8) -> Void = {
-#if _endian(little)
-        buffer = buffer | (UInt64($0) << (UInt64(i) * 8))
-#else
-        buffer = buffer | (UInt64($0) << (UInt64(7-i) * 8))
-#endif
-        i += 1
-      }
-      UTF8.encode(unicodeScalar, into: sink)
-      return body(UnsafeBufferPointer(
-        start: UnsafePointer(Builtin.addressof(&buffer)),
-        count: i))
+      return unicodeScalar.withUTF8CodeUnits { body($0) }
```

Comment:
This is using the internal [`Unicode.Scalar.withUTF8CodeUnits(_:)`][1] method, which has been ABI-public since Swift 5.0 — I assume it will also be available in the bundled stdlib (e.g. when targeting macOS 10.9).

[1]: <https://github.com/apple/swift/blob/swift-5.0-branch/stdlib/public/core/UnicodeScalar.swift#L470-L482>

Reply:
Yep, those were built from the same sources!

(It's unfortunate that this non-public entry point is missing its leading underscore.)

Reply:
(There was also a [fix in Swift 5.1][1] for big-endian platforms.)

[1]: <https://github.com/apple/swift/blame/swift-5.1-branch/stdlib/public/core/UnicodeScalar.swift#L522-L524>